src/io/dirops.c

```diff
@@ -33,79 +33,33 @@ static char * UnicodeToUTF8(const wchar_t *str)
 
      return result;
 }
+#endif
 
-static int mkdir_p(MVMThreadContext *tc, wchar_t *pathname, MVMint64 mode) {
-    wchar_t *p = pathname, ch;
-#else
 static int mkdir_p(MVMThreadContext *tc, char *pathname, MVMint64 mode) {
     char *p = pathname, ch;
     uv_fs_t req;
     int mkdir_error = 0;
-#endif
     int created = 0;
 
     for (;; ++p)
         if (!*p || IS_SLASH(*p)) {
             ch = *p;
             *p  = '\0';
-#ifdef _WIN32
-            created = CreateDirectoryW(pathname, NULL);
-#else
             created = (mkdir(pathname, mode) == 0
                        || ( (mkdir_error = errno) == EEXIST
                        && uv_fs_stat(NULL, &req, pathname, NULL) == 0
-                       && S_ISDIR(req.statbuf.st_mode)));
-#endif
+                       ));
             if (!(*p = ch)) break;
         }
 
-#ifndef _WIN32
-    errno = mkdir_error; /* error reporting should use errno of latest mkdir */
-#endif
     if (!created) return -1;
 
     return 0;
 }
 
 /* Create a directory recursively. */
 void MVM_dir_mkdir(MVMThreadContext *tc, MVMString *path, MVMint64 mode) {
-    char * const pathname = MVM_string_utf8_c8_encode_C_string(tc, path);
-
-#ifdef _WIN32
```

Comment:
We delete all of this because it is trying to do its own path prefix stuff with `\\?\`, and doesn't work correctly when `str_len > MAX_PATH` and the path is already prefixed with `\\?\`